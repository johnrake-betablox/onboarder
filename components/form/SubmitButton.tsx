import { classNames } from "@/utils/classnames";
import LoadingDots from "../LoadingDots";

export default function SubmitButton({
  loading = false,
  text = "Save",
}: {
  loading?: boolean;
  text?: string;
}) {
  return (
    <button
      type="submit"
      className={classNames(
        loading
          ? "cursor-not-allowed bg-gray-200 text-black opacity-80"
          : "bg-green-600 text-white hover:bg-green-500",
        "rounded  px-3 py-2 font-semibold",
      )}
      disabled={loading}
    >
      {loading ? <LoadingDots /> : text}
    </button>
  );
}
