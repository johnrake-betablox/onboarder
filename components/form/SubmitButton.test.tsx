import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import SubmitButton from "./SubmitButton";

test("renders a snapshot", () => {
  const { asFragment, getByText } = render(<SubmitButton />);
  expect(getByText("Save")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});

test("renders with optional text", () => {
  const { asFragment, getByText } = render(
    <SubmitButton text="Create Account" />,
  );
  expect(getByText("Create Account")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
