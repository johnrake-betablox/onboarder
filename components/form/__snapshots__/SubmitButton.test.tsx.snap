// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<DocumentFragment>
  <button
    class="bg-green-600 text-white hover:bg-green-500 rounded  px-3 py-2 font-semibold"
    type="submit"
  >
    Save
  </button>
</DocumentFragment>
`;

exports[`renders with optional text 1`] = `
<DocumentFragment>
  <button
    class="bg-green-600 text-white hover:bg-green-500 rounded  px-3 py-2 font-semibold"
    type="submit"
  >
    Create Account
  </button>
</DocumentFragment>
`;
